Documentation/design-docs/host-startup-hook.md

```diff
@@ -93,4 +94,119 @@ namespace SharedHostPolicy
         }
     }
 }
-```
\ No newline at end of file
+```
+
+## Error handling details
+
+Problems with the startup hook should be fairly straightforward to
+diagnose. All of these exceptionts will contain the startup hook path
+(`System.StartupHookProvider.ProcessStartupHooks`) on the stack
+trace. They fall into the following categories:
+
+- Errors detected eagerly, with exceptions thrown before the execution
+  of any startup hook.
+
+-- Invalid syntax. This will throw an `ArgumentException` with "The
```

Comment:
Documenting the messages in the exception isn't a good idea. It encourages people to do weird things and write tests that check error messages - which is very brittle.

Reply:
Ok, I'll remove the exception messages from the spec.